src/board.py

```diff
@@ -117,18 +96,19 @@ def __repr__(self) -> str:
             "\n▐\033[7m  A B C D E F G H  \033[0m▌\n\n"
         ).replace("None", " ")
 
-    def __setitem__(self, square: Square | Indices | None, piece: Piece | None):
+    def __setitem__(self, square: Square | str | None, piece: Piece | None):
         """Add a piece to a square.
 
         Args:
             square: The rank and file of the square.
             piece: The piece to be placed on the square.
         """
         if square:
-            i, j = self._indices(square) if isinstance(square, Square) else square
-            self._board[i][j] = piece
+            if isinstance(square, str):
```

Comment:
No usage of `_square` here? If not, delete method.
Alternatively, do `square = Square(square)` even when `square` is a `Square`, it should work.

Reply:
Wait, what is that. I do not recognize it, that shouldn't be there. :D